code/modules/almayer/weaponhits.dm

```diff
@@ -69,21 +88,91 @@
 				for(var/turf/picked_atom in location)
 					if(prob(hitchance))
 						cell_explosion(picked_atom, 600, 600, EXPLOSION_FALLOFF_SHAPE_EXPONENTIAL, null, antishiprailgun_cause_data)
+						GLOB.ship_health_vars["ship_hull_health"] = GLOB.ship_health_vars["ship_hull_health"] - WEAPON_RAILGUN_DAMAGE
 						shakeship(5, 5, FALSE, FALSE)
 						confirmedhit += 1
+						logtotalhits += 1
 				if(confirmedhit > 0)
 					weaponhits_effects(WEAPON_RAILGUN)
 				if(confirmedhit < 1)
 					weaponhits_effects(WEAPON_RAILGUN, TRUE)
+				GLOB.ship_hits_tally["times_hit_railgun"] += confirmedhit
 
 			else if(salvo == FALSE)
 				if(prob(hitchance))
 					cell_explosion(location, 600, 600, EXPLOSION_FALLOFF_SHAPE_EXPONENTIAL, null, antishiprailgun_cause_data)
+					GLOB.ship_health_vars["ship_hull_health"] = GLOB.ship_health_vars["ship_hull_health"] - WEAPON_RAILGUN_DAMAGE
 					shakeship(5, 5, FALSE, FALSE)
 					weaponhits_effects(WEAPON_RAILGUN)
+					GLOB.ship_hits_tally["times_hit_railgun"] += 1
+					logtotalhits += 1
 				else
 					weaponhits_effects(WEAPON_RAILGUN, TRUE)
 
+		if(WEAPON_ODC)
+			var/datum/cause_data/orbitaldefensecannon_cause_data = create_cause_data("Orbital Defense Shell")
+			var/hitchance = HIT_CHANCE_CHEAT
+			if(point_defense == TRUE)
+				hitchance = HIT_CHANCE_STANDARD
+			if(salvo == TRUE)
+				var/confirmedhit
+				for(var/turf/picked_atom in location)
+					if(prob(hitchance))
+						cell_explosion(picked_atom, 850, 6, EXPLOSION_FALLOFF_SHAPE_EXPONENTIAL, null, orbitaldefensecannon_cause_data)
+						GLOB.ship_health_vars["ship_hull_health"] = GLOB.ship_health_vars["ship_hull_health"] - WEAPON_ODC_DAMAGE
+						shakeship(10, 10, TRUE, FALSE)
+						confirmedhit += 1
+						logtotalhits += 1
+				if(confirmedhit > 0)
+					weaponhits_effects(WEAPON_ODC)
+				if(confirmedhit < 1)
+					weaponhits_effects(WEAPON_ODC, TRUE)
+				GLOB.ship_hits_tally["times_hit_odc"] += confirmedhit
```

Comment:
```suggestion
						confirmedhit++
						logtotalhits++
					weaponhits_effects(WEAPON_ODC, confirmedhit < 1)
				GLOB.ship_hits_tally["times_hit_odc"] += confirmedhit
```
If you don't intend to ever have multiple types of weapons hitting in one call, then you don't even need confirmedhit and instead could use logtotalhits.